doc/lsst.ap.pipe/apdb.rst

```diff
@@ -14,7 +14,9 @@ Setting up the Alert Production Database for ap_pipe
 
 .. |ap_pipe| replace:: :command:`ap_pipe.py`
 
-|ap_pipe| needs an existing Alert Production Database (APDB) in which to store its results.
+The Alert Production Pipeline, as represented by :lsst-task:`lsst.ap.pipe.ApPipeTask` and executed by |ap_pipe|, delegates saving and loading DIASources and DIAObjects to its ``diaPipe`` subtask.
+In principle, different implementations of ``diaPipe`` can be selected in defined in `ApPipeTask`'s configuration (by setting :lsst-config-field:`lsst.ap.pipe.ap_pipe.ApPipeConfig.diaPipe`), but the default choice --- :lsst-task:`lsst.ap.association.DiaPipelineTask` --- is expected to be appropriate for most uses.
+`~lsst.ap.association.DiaPipelineTask` uses a database --- known as the Alert Production Database, or APDB --- to store its results.
```

Comment:
This is a good point, and I wondered about it when writing the text.

My foresight is not as good as you imply it might be. However, given that the APDB usage is isolated in a retargetable subtask, I'm a bit leery of talking about it as being fundamental to the design of the pipeline. Given your comments, though, maybe that's a worthwhile simplification. I'll do some redrafting.

Reply:
That's a good point. We don't want `DiaPipelineTask` to be a subtask; we wrote it so that it would be a standalone `PipelineTask` in Gen 3, so there's not even an implicit contract that any alternative has to provide database support.

However, I don't think we'll be able to write a truly generic description of how this should work until [DM-22663](https://jira.lsstcorp.org/browse/DM-22663), so I think the simplification is worthwhile for now.